src/openloco/window.cpp

```diff
@@ -117,6 +117,40 @@ namespace openloco::ui
             viewports[1]->render(dpi);
     }
 
+    // 0x0045FCE6
+    // Input:
+    // regs.ax:  x
+    // regs.bx:  y
+    // regs.bp:  z
+    // Output:
+    // {x: regs.ax, y: regs.bx}
+    // Note: function seems to be similar to screen_get_map_xy_with_z in openrct2
+    map::map_pos sub_45FCE6(int16_t x, int16_t y, int16_t z)
+    {
+        window* w = WindowManager::findAt(x, y);
+        if (w != nullptr)
+        {
+            viewport* vp = w->viewports[0];
+            if (vp != nullptr && x >= vp->x && x < vp->x + vp->width && y >= vp->y && y < vp->y + vp->height)
+            {
+                map::map_pos position = viewport_coord_to_map_coord(
+                    ((x - vp->x) << vp->zoom) + vp->view_x,
```

Comment:
create a `ui_to_map` viewport function for this

Reply:
There is already a `ui_to_map` function defined which has a different implementation. Can you look at it?

Reply:
In openrct2 we have this function: `screen_coord_to_viewport_coord` which is similar what we need.

Reply:
There is a `map_to_ui` function in the codebase but this would be the reverse `ui_to_map`. Yes it would be the same as the OpenRCT2 `screen_coord_to_viewport_coord` function.

Reply:
Ok, good. Sorry, I didn't notice that little difference. I will make that function.

Reply:
Ok, implemented now.